tests/Console/Command/DeletePlaceTest.php

```diff
@@ -0,0 +1,144 @@
+
+        $command = new DeletePlace(
+            $this->commandBus,
+            $this->eventRelationsRepository,
+            $this->placeDocumentRepository
+        );
+
+        $application = new Application();
+        $application->add($command);
+
+        $command = $application->find('place:delete');
+        $this->commandTester = new CommandTester($command);
+    }
+
+    public function test_execute_missing_arguments(): void
+    {
+        $this->commandTester->execute([
+            'place-uuid' => null,
+            'canonical-uuid' => null,
+        ]);
+
+        $output = $this->commandTester->getDisplay();
+        $this->assertStringContainsString('Missing argument, the correct syntax is', $output);
+    }
+
+    public function test_execute_place_does_not_exist(): void
+    {
+        $placeUuid = Uuid::uuid4()->toString();
+        $canonicalUuid = Uuid::uuid4()->toString();
+        $this->placeDocumentRepository
+            ->method('fetch')
+            ->willReturnCallback(function ($id) use ($placeUuid, $canonicalUuid) {
+                if ($id === $canonicalUuid) {
+                    return new JsonDocument($id, json_encode([]));
+                }
+
+                if ($id === $placeUuid) {
+                    throw new DocumentDoesNotExist('Place does not exist');
+                }
+            });
+
+
+        $this->commandTester->execute([
+            'place-uuid' => $placeUuid,
+            'canonical-uuid' => $canonicalUuid,
+        ]);
+
+        $output = $this->commandTester->getDisplay();
+        $this->assertStringContainsString('Place does not exist', $output);
+    }
+
+    public function test_execute_canonical_does_not_exist(): void
+    {
+        $placeUuid = Uuid::uuid4()->toString();
+        $canonicalUuid = Uuid::uuid4()->toString();
+        $this->placeDocumentRepository
+            ->method('fetch')
+            ->with($canonicalUuid)
+            ->willThrowException(new DocumentDoesNotExist('Place does not exist'));
+
+        $this->commandTester->execute([
+            'place-uuid' => $placeUuid,
+            'canonical-uuid' => $canonicalUuid,
+        ]);
+
+        $output = $this->commandTester->getDisplay();
+        $this->assertStringContainsString('Canonical place does not exist', $output);
+    }
+
+    public function test_execute_successfully_dispatches_commands(): void
+    {
+        $eventId1 = Uuid::uuid4()->toString();
+        $eventId2 = Uuid::uuid4()->toString();
+        $placeUuidToDelete = 'place-uuid';
+        $canonicalUuid = new LocationId(Uuid::uuid4()->toString());
+
+        $this->placeDocumentRepository
+            ->method('fetch')
+            ->willReturn(new JsonDocument(Uuid::uuid4()->toString(), json_encode([])));
+
+        $this->eventRelationsRepository
+            ->method('getEventsLocatedAtPlace')
+            ->willReturn([$eventId1, $eventId2]);
+
+        $dispatchedCommands = [];
+        $this->commandBus
```

Comment:
There is a special class to handle this, the class is called `TraceableCommandBus`